typescript/token/test/HypERC20.t.sol

```diff
@@ -0,0 +1,366 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity ^0.8.13;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+import "forge-std/Test.sol";
+
+import {TypeCasts} from "@hyperlane-xyz/core/contracts/libs/TypeCasts.sol";
+import {TestMailbox} from "@hyperlane-xyz/core/contracts/test/TestMailbox.sol";
+import {TestPostDispatchHook} from "@hyperlane-xyz/core/contracts/test/TestPostDispatchHook.sol";
+import {TestInterchainGasPaymaster} from "@hyperlane-xyz/core/contracts/test/TestInterchainGasPaymaster.sol";
+import {GasRouter} from "@hyperlane-xyz/core/contracts/GasRouter.sol";
+
+import {ERC20Test} from "../contracts/test/ERC20Test.sol";
+import {HypERC20} from "../contracts/HypERC20.sol";
+import {HypERC20Collateral} from "../contracts/HypERC20Collateral.sol";
+import {HypNative} from "../contracts/HypNative.sol";
+import {TokenRouter} from "../contracts/libs/TokenRouter.sol";
+
+abstract contract HypTokenTest is Test {
+    using TypeCasts for address;
+    uint32 internal constant ORIGIN = 11;
+    uint32 internal constant DESTINATION = 12;
+    uint8 internal constant DECIMALS = 18;
+    uint256 internal constant TOTAL_SUPPLY = 1_000_000e18;
+    uint256 internal REQUIRED_VALUE; // initialized in setUp
+    uint256 internal constant GAS_LIMIT = 10_000;
+    uint256 internal IGP_GAS_PRICE; // initialized in test
+    uint256 internal constant TRANSFER_AMT = 100e18;
+    string internal constant NAME = "HyperlaneInu";
+    string internal constant SYMBOL = "HYP";
+    address internal constant ALICE = address(0x1);
+    address internal constant BOB = address(0x2);
```

Comment:
all of this could probably be fuzzed

Reply:
probably, but this is for the token standards and outside the scope of hyp- contracts